### PR TITLE
Sgrouard/sanity checks updated

### PR DESCRIPTION
**Pull request by SimonGrouard**
Updated sanity check files
It includes the more granular cell categories, as well as the debugging of sanity check 3.

### CODE REVIEW

File: notebooks/sanity_checks_utils.py [hunk 1]
```diff
@@ -62,46 +63,46 @@
...
+        "T": ["MAIT","T_CD4/CD8","Tgd_CRTAM+","Trm_Tgd"],
         "NK": ["NK_CD16+", "NK_CD56bright_CD16-"],
-        "ICL3": ["ILC3"],
         "DC": ["DC1", "DC2", "migDC", "pDC"],
         "Mast": ["Mast cells"],
         "Red_blood": ["Erythroid"],
         "Bone_marrow": ["Megakaryocytes"],
         "Non-differentiated": ["Progenitor"],
-        "To remove": ["Cycling", "T/B doublets", "Cycling T&NK"],
+        "To remove":  ["Cycling", "T/B doublets", "Cycling T&NK",
```
Comment (khalilouardini) [inline notebooks/sanity_checks_utils.py:83]: Is this going to be updated after your conversation with Sabrina ?

**Review by khalilouardini — COMMENTED** [review]
Thanks @SimonGrouard ! i have a few comments.
Also, in the part where we do the train split - how are we sure that the test set wasn't used in the building of the siganture matrix, since we split randomly ?

**Comment by SimonGrouard** [comment]
> Thanks @SimonGrouard ! i have a few comments. Also, in the part where we do the train split - how are we sure that the test set wasn't used in the building of the siganture matrix, since we split randomly ?

We do a random split with a fixed random state of 42. The same random state is also used to create the train indices / test indices which we use to create the signature matrix.

File: notebooks/deconvo_sanity_checks.py [hunk 2]
```diff
@@ -133,8 +137,12 @@
 # %%
 # Sanity check 2: Pseudobulks from predefined fractions → correlation between the
 # fractions of selected cells (“ground truth”)  and the estimated fractions
+# nb This task is too easy because the pseudo-bulk have the same cell fractions than the
+# training dataset on which was created the signature matrix.
 
-n_sample = 100
+n_sample = 100 # When using 1000 to create the pseudo-bulks, all n_sample pseudo-bulks
```
[the remark this answers is not in the document]
Comment (SimonGrouard) [reply notebooks/deconvo_sanity_checks.py:143]: Hum not so far, but maybe we should try it

File: notebooks/sanity_checks_utils.py (hunk 1 above)
Comment (SimonGrouard) [reply notebooks/sanity_checks_utils.py:83]: Yes, just CD8 T -> CD8T and CD4 T -> CD4T

File: notebooks/deconvo_sanity_checks.py [hunk 3]
```diff
@@ -28,10 +28,12 @@
 signature_almudena = read_almudena_signature(
     "/home/owkin/project/Almudena/Output/Crosstiss_Immune_norm/CTI.txt"
 )  # it is the normalised one (using adata.X and not adata.raw.X, to match this code)
+# TO ADD : the new signature with granular updated cell types
 
 
 # %%
 # cell type categories
+# TO ADD : possibility of crosstissue_granular_updated
```
Comment (khalilouardini) [inline notebooks/deconvo_sanity_checks.py:36]: will be added in another PR ?

File: notebooks/deconvo_sanity_checks.py [hunk 4]
```diff
@@ -28,10 +28,12 @@
 signature_almudena = read_almudena_signature(
     "/home/owkin/project/Almudena/Output/Crosstiss_Immune_norm/CTI.txt"
 )  # it is the normalised one (using adata.X and not adata.raw.X, to match this code)
+# TO ADD : the new signature with granular updated cell types
```
Comment (khalilouardini) [inline notebooks/deconvo_sanity_checks.py:32]: will be added in another PR ?

File: notebooks/deconvo_sanity_checks.py [hunk 5]
```diff
@@ -63,6 +65,9 @@
     random_state=42,
 )
 adata = adata[cell_types_test, :]
+# WARNING. The following line appears in the R sanity checks script:
+# scRNseq_test <- NormalizeData(object = scRNseq_test, normalization.method = "RC",scale.factor = 10000)
+# Should we thus add it here ? (on the raw counts I guess)
```
Comment (khalilouardini) [inline notebooks/deconvo_sanity_checks.py:71]: Not sure I understand this comment. Thus normalization is applied before the computation of the signature matrix int he R script ?

Comment (khalilouardini) [reply notebooks/deconvo_sanity_checks.py:71]: do we know what "RC" means ? in our case, the normalization is done in scanpy= `sc.pp.normalize_total(adata, target_sum=1e4)`

**Review by khalilouardini — COMMENTED** [review]
few minor comments about the comments lol.
Otherwise lgtm!

File: notebooks/deconvo_sanity_checks.py (hunk 3 above)
Comment (SimonGrouard) [reply notebooks/deconvo_sanity_checks.py:36]: Yes! I guess it can be done in the same one as when integrating MixUpVI to the sanity checks

File: notebooks/deconvo_sanity_checks.py (hunk 4 above)
Comment (SimonGrouard) [reply notebooks/deconvo_sanity_checks.py:32]: Same answer as above, yes

File: notebooks/deconvo_sanity_checks.py (hunk 5 above)
Comment (khalilouardini) [reply notebooks/deconvo_sanity_checks.py:71]: Ok. Is this normalization is done in the R scripts that builds the signature matrix? if yes, why should we add it here ?

Comment (khalilouardini) [reply notebooks/deconvo_sanity_checks.py:71]: (To normalize the pseudobulk dataset with the same normalization?)

Comment (SimonGrouard) [reply notebooks/deconvo_sanity_checks.py:71]: Normally, the same R scripts are used to create the signature matrix and then perform the sanity checks. Therefore, the normalization is applied on the test data used for the sanity checks.
